src/components/NavButtonWithTooltip/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import { Tooltip, TooltipProps } from "antd";
+
+import {
+    NAV_BAR_TOOLTIP_OFFSET,
+    TOOLTIP_COLOR,
+    TOOLTIP_DELAY,
+} from "../../constants";
+import NavButton, { NavButtonProps } from "../NavButton";
+
+interface TooltipText {
+    default: string;
+    disabled?: string;
+}
+interface NavButtonWithTooltipProps extends NavButtonProps {
+    tooltipText?: TooltipText;
+    tooltipPlacement?: TooltipProps["placement"];
+}
+
+const NavButtonWithTooltip: React.FC<NavButtonWithTooltipProps> = ({
+    tooltipText = { default: "" },
```

Comment:
```suggestion
    tooltipText = { default: "", disabled: "" },
```

Reply:
Oh this is great, with this and checking `if (tooltipVisible)` like you suggested we can actually get rid of most of the conditionals in both the `useEffect` and `getRenderTooltipText`